storage/pebble/procedure/cluster.go

```diff
@@ -131,13 +134,13 @@ func FinalizeClusterBlock(blockID flow.Identifier) func(*badger.Txn) error {
 		}
 
 		// insert block view -> ID mapping
-		err = operation.IndexClusterBlockHeight(chainID, header.Height, header.ID())(tx)
+		err = operation.IndexClusterBlockHeight(chainID, header.Height, header.ID())(w)
 		if err != nil {
 			return fmt.Errorf("could not insert view->ID mapping: %w", err)
 		}
 
 		// update the finalized boundary
-		err = operation.UpdateClusterFinalizedHeight(chainID, header.Height)(tx)
+		err = operation.UpdateClusterFinalizedHeight(chainID, header.Height)(w)
```

Comment:
## ⚠️ 

this is always vulnerable to stale writes in case of concurrent access, which would corrupt the state. See [my related comment](https://github.com/onflow/flow-go/pull/6197#discussion_r1715978123) for details